src/modules/iotjs_module_gpio.c

```diff
@@ -198,13 +206,17 @@ jerry_value_t gpio_do_write_or_writesync(const jerry_value_t jfunc,
   } else if (jerry_value_is_boolean(jargv[0])) {
     value = jerry_get_boolean_value(jargv[0]);
   } else {
-    const char* js_create_error_arg;
-    if (gpio_op == IOTJS_GPIO_WRITE) {
-      js_create_error_arg = iotjs_periph_error_str(kGpioOpWrite);
-    } else {
-      js_create_error_arg = "GPIO WriteSync Error - Wrong argument type";
+    const jerry_value_t jcallback = JS_GET_ARG_IF_EXIST(1, function);
+    if (gpio_op == IOTJS_GPIO_WRITE && !jerry_value_is_null(jcallback)) {
+      const char* error_msg = iotjs_periph_error_str(kGpioOpWrite);
+      jerry_value_t error_str = jerry_create_string((jerry_char_t*)error_msg);
+      jerry_value_t args[] = { error_str };
+      iotjs_invoke_callback(JS_GET_ARG_IF_EXIST(1, function), jthis, args, 1);
```

Comment:
This code can be simplified a bit:
```c
const jerry_value_t jcallback = JS_GET_ARG_IF_EXIST(1, function);
if (gpio_op == IOTJS_GPIO_WRITE && !jerry_value_is_null(jcallback)) {
  const char* error_msg = iotjs_periph_error_str(kGpioOpWrite);
  jerry_value_t error_str = jerry_create_string((jerry_char_t*)error_msg);
  iotjs_invoke_callback(jcallback, jthis, &error_str, 1);
  jerry_release_value(error_str);
 
 return jerry_create_undefined();
}
```
Please add the test case that @hs0225 mentioned (`gpio.write({})`) to `test_gpio_api.js`.